### ISSUE TITLE
Manually created geoarrow table support in ScatterplotLayer

### ISSUE THREAD

**Issue by ingenieroariel** [developmentseed/lonboard#152]
I was able to load 20 million polygons in lonboard. It was amazing! Now I am trying to figure out how to load 60 million points without having to use GeoPandas but I keep hitting code paths that expect either an interleaved list or paths that go back to numpy or paths that expect a byte like object from C.

Here is roughly what I am trying:

```
import lonboard
import gzip
import geoarrow.pyarrow as ga
import pyarrow.csv as pv

with gzip.open("/Users/x/data/points_s2_level_4_gzip/397_buildings.csv.gz") as fp:
        table = pv.read_csv(fp)

points = ga.point().from_geobuffers(None, table["latitude"], y=table["longitude"])

geoarrow_schema = pa.schema([pa.field("geometry", points.type, metadata={b"ARROW:extension:name": b"geoarrow.point"})])

point_table = pa.Table.from_arrays([points], schema=geoarrow_schema)
point_table.schema.field("geometry").metadata.get(b"ARROW:extension:name")
map_ = lonboard.ScatterplotLayer(table=point_table)
```

**Comment by kylebarron** [comment]
lonboard is designed to support geoarrow data directly, though so far it's mostly been tested on geopandas input. Can you say what the error is? Everything that expects a numpy array also allows a pyarrow array https://developmentseed.org/lonboard/api/traits/#lonboard.traits.ColorAccessor

**Comment by ingenieroariel** [comment]
I tried a few things and lost that stack trace. Logged the first issue with just going with arrow tables directly upstream:

https://github.com/geoarrow/geoarrow-python/issues/33

**Comment by ingenieroariel** [comment]
```python
import lonboard
import gzip
import geoarrow.pyarrow as ga
import pyarrow.csv as pv
import pyarrow as pa

with gzip.open("/Users/x/data/points_s2_level_4_gzip/397_buildings.csv.gz") as fp:
        table = pv.read_csv(fp)
points = ga.point().from_geobuffers(None, table["latitude"].combine_chunks().to_numpy(), y=table["longitude"].combine_chunks().to_numpy())
geoarrow_schema = pa.schema([pa.field("geometry", points.type, metadata={b"ARROW:extension:name": b"geoarrow.point"})])
point_table = pa.Table.from_arrays([points], schema=geoarrow_schema)
```

map_ = lonboard.ScatterplotLayer(table=point_table)
map_
```

File ~/tmp/lib/python3.11/site-packages/lonboard/viewport.py:27, in get_bbox_center(table)
     24 geom_field = table.schema.field(geom_col_idx)
     25 geom_col = table.column(geom_col_idx)
---> 27 bbox = total_bounds(geom_field, geom_col)
     28 centroid = weighted_centroid(geom_field, geom_col)
     30 return bbox, centroid

File ~/tmp/lib/python3.11/site-packages/lonboard/geoarrow/ops/bbox.py:42, in total_bounds(field, column)
     39 extension_type_name = field.metadata[b"ARROW:extension:name"]
     41 if extension_type_name == EXTENSION_NAME.POINT:
---> 42     return _total_bounds_nest_0(column)
     44 if extension_type_name in [EXTENSION_NAME.LINESTRING, EXTENSION_NAME.MULTIPOINT]:
     45     return _total_bounds_nest_1(column)

File ~/tmp/lib/python3.11/site-packages/lonboard/geoarrow/ops/bbox.py:67, in _total_bounds_nest_0(column)
     65 for chunk in column.chunks:
     66     coords = chunk
---> 67     bbox.update(_coords_bbox(coords))
     69 return bbox

File ~/tmp/lib/python3.11/site-packages/lonboard/geoarrow/ops/bbox.py:57, in _coords_bbox(arr)
     56 def _coords_bbox(arr: pa.FixedSizeListArray) -> Bbox:
---> 57     np_arr = arr.flatten().to_numpy().reshape(-1, arr.type.list_size)
     58     min_vals = np.min(np_arr, axis=0)
     59     max_vals = np.max(np_arr, axis=0)

AttributeError: 'PointArray' object has no attribute 'flatten'
```

**Comment by kylebarron** [comment]
Oh well it's integrated with _pyarrow_, it's not integrated with whatever new types geoarrow-python is making. If you have a chunked array with the storage type of `PointArray` (i.e. a `FixedSizeListArray`) it'll work

**Comment by ingenieroariel** [comment]
Thanks for that comment, I grepped for FixedSizeListArray on this repo and found how to implement your recommendation. 

Closing the issue and leaving the code below for others:

```python
import gzip
import pyarrow.csv as pv
import pyarrow as pa
import lonboard

with gzip.open("/Users/x/data/points_s2_level_4_gzip/397_buildings.csv.gz") as fp:
        table = pv.read_csv(fp)
coords = np.array([table["latitude"].to_numpy(), table["longitude"].to_numpy()]).T.flatten()
parr = pa.FixedSizeListArray.from_arrays(coords, len(coords))
field = pa.field(
            "geometry",
            parr.type,
            nullable=True,
            metadata={"ARROW:extension:name": "geoarrow.point"},
    )
schema = pa.schema([field])
table = pa.Table.from_arrays([parr], schema=schema)

map_ = lonboard.ScatterplotLayer(table=table)
```

The ScatterplotLayer is created successfully!

**Comment by kylebarron** [comment]
Did it work with 60 million points?

**Comment by ingenieroariel** [comment]
Nope. Got a javascript error, only 20 gb of ram was used (from 128GB), some 100mb of swap was created and javascript errored out. I tried with 20M of those points and it broke too.

My feeling is that the polygon workflow I am using is better vectorized than the point workflow I am following so more tests are due.

This is all so new, so I decided to try the largest and report that instead of reporting what does not work.

New frontiers, lonboard ride down the mountain.
